packages/ckb-sdk-core/src/signWitnesses.ts

```diff
@@ -46,30 +53,59 @@ const signWitnesses: SignWitnesses = (key: SignatureProvider | Map<LockHash, Sig
   if (!witnesses.length) throw new Error('Witnesses is empty')
 
   if (isMap(key)) {
+    if (!inputCells.length) {
+      throw new Error(`Cell shouldn't be empty when key is Map`)
+    }
     const rawWitnesses = witnesses
     const restWitnesses = witnesses.slice(inputCells.length)
     const groupedScripts = groupScripts(inputCells)
-
-    groupedScripts.forEach((indices, lockhash) => {
+    const lockhashes = [...groupedScripts.keys()]
+    for (let i = 0; i < lockhashes.length; i++) {
+      const lockhash = lockhashes[i];
       const sk = key.get(lockhash)
       if (!sk) {
         if (!skipMissingKeys) {
           throw new Error(`The signature provider to sign lockhash ${lockhash} is not found`)
         } else {
-          return
+          continue
         }
       }
 
+      const indices = groupedScripts.get(lockhash)!
       const ws = [...indices.map(idx => witnesses[idx]), ...restWitnesses]
-
-      const witnessIncludeSignature = signWitnessGroup(sk, transactionHash, ws)[0]
-      rawWitnesses[indices[0]] = witnessIncludeSignature
-    })
+      let signStatus = SignStatus.Signed
```

Comment:
default value should be `unsigned`

Reply:
If it isn't multisig, it will be `Signed` after sign. So the default value is `Signed`

Reply:
We can put it in the `multisig` arm if it's only used in there and set its default value `unsigned`